### DISCUSSION TITLE
A new approach to mary dashboards and alerts or "don't link, search"

### DISCUSSION THREAD

**Discussion by yuri-tceretian** [grafana/grafana#45476] [Dashboards, Panels & Visualizations]
### Background
Before Grafana 8 alerts were an integral part of a panel\dashboard: to create an alert users had to create a dashboard, panel, and then alert. Grafana 8 brought a lot of improvements in alerting: it separated alerts from dashboards, started to support multidimensional alert rules. Roughly, a single query can produce many alert instances (a.k.a dimensions).  Every alert instance has a unique set of labels (name + value) which is the superset of the labels declared by the alert rule (because the result of querying a data source can contain more labels that are not declared by the alert rule). 
Although alerts now can exist without a dashboard\panel, Grafana still provides a mechanism to assign an alert rule to a panel. Currently, it can be done by declaring special annotations for the alert rule: DashboardUID and PanelID. This allows simulating the backward compatibility with legacy alerting. However, this does not let users leverage all new features that the new Grafana alerting engine offers:
- limit the usage of multi-dimensional alerts. It does not make sense to link a multi-dimensional alert rule to a panel that shows only one dimension
- dashboard variables are not applied to annotations created by the rule evaluation

### Proposal

TL;DR; "do not link, search" 

Instead of assigning an "alert rule" to a panel, we make the panel search for an "alert instance". This is important to understand the difference between "alert rule" and "alert instance": the latter is the result of the former, and in the case of the multi-dimensional alerts "alert instance" is a single dimension. In the panel configuration, a user will have to declare one or multiple matching rules. All alert instances that match the rules will be displayed in the panel. Matchers can use dashboard variables.
Pros:
- loose coupling between alerts and dashboards\panels
- the ability to display alerts for multiple metrics of a single object (imagine a panel that shows the throughput of an app on a single server and displays alerts for the server's CPU, Memory, IO etc, where every metric is served by a separate alert rule).

Cons:
- makes authorization mechanisms a little bit more complex than in the current implementation

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
